### ISSUE TITLE
Consider being defensive wrt. labels which contain `:`

### ISSUE THREAD

**Issue by ktoso** [apple/swift-statsd-client#21]
This is a bug which breaks using swift-statsd with Vapor.

It's debatable where this should be fixed (if label sanitization could be part of swift metrics APIs or not), but following prior discussions and art (how we fixed it in  swift-prometheus via "LabelSanitizers" https://github.com/MrLotU/SwiftPrometheus/commit/938b2a29be42e756959175315d87c978d9f8e64e ), let's perhaps fix it in the backend here.

--- 

Details:

Vapor has built in metrics and emits them here https://github.com/vapor/vapor/blob/633b2a81f547383aa6becb6ba1ad011dc7174779/Sources/Vapor/Responder/DefaultResponder.swift#L83

It does so for each route (very good):

```
            self.updateMetrics(
                for: request,
                path: path,
                startTime: startTime,
                statusCode: status.code
            )
```

and it identifies them by the route description (the problem):

```
            path = cachedRoute.route.description
```

since `route.description` would be something like: `/users/:uuid`

thus the reported metric would be:

```swift
/users/:uuid:111|c
// MetricName:value|type|@sample_rate|#tag1:
```

so the `:` of the uuid breaks the format.

---

Long story short, we need label sanitizers the same way as swift-prom does.

I'll PR one here and we can at some point consider if it's a general concept that should exist for all swift-metrics backends.
